service/csv/csv.go

```diff
@@ -0,0 +1,198 @@
+// go:generate mockgen -source=csv.go -destination=mock/csv_mock.go -package mock
+package csvservice
+
+import (
+	"encoding/csv"
+	"errors"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+
+	"pokeapi/model"
+)
+
+const pathFile = "./csv/pokemon.csv"
+
+type CsvService struct{}
+
+type NewCsvService interface {
+	GetPokemons() ([]model.Pokemon, *model.Error)
+	GetPokemon(pokemonId int) (model.Pokemon, *model.Error)
+	SavePokemons(*[]model.SinglePokeExternal) *model.Error
+}
+
+func New() *CsvService {
+	return &CsvService{}
+}
+
+func Open(path string) (*os.File, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, errors.New("There was an error opening the file")
+	}
+	return f, nil
+}
+
+func OpenAndWrite(path string) (*os.File, error) {
+	f, err := os.OpenFile(path, os.O_RDONLY|os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
+	if err != nil {
+		return nil, errors.New("There was an error opening the file")
+	}
+	return f, nil
+}
+
+func Read(f *os.File) ([]model.Pokemon, *model.Error) {
+
+	reader := csv.NewReader(f)
+	reader.Comma = ','
+	reader.Comment = '#'
+	reader.FieldsPerRecord = -1
+
+	var pokemons []model.Pokemon = nil
+	for {
+		line, err := reader.Read()
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			err := model.Error{
+				Code:    http.StatusInternalServerError,
+				Message: err.Error(),
+			}
+			return nil, &err
+		}
+		tempPokemon := model.Pokemon{
+			Name: line[1],
+			URL:  line[2],
+		}
+
+		if line[0] != "" {
+			id, err := strconv.Atoi(line[0])
+			if err != nil {
+				err := model.Error{
```

Comment:
You would probably like to return your custom error on the upper layer before this one. I would recommend returning this specific error to your usecase, log it there, and then let the usecase return your custom error. 
This is important to now exactly where and how the program flow failed.

This applies for all of the errors in your services